File: c 풀이/H4.c
#include <stdio.h>

void func(int k, int n)
{
	int time = 0;
	int bomb = k;
	int t; 
	char a='A';

	for (int i = 0; i < n; i++) {
		scanf(" %d %c", &t, &a);
		time += t;
		if (time >= 210) break;
		else {
			a == 'T' ? (bomb += 1) : (bomb += 0) ;
		}
	}
	printf("%d", bomb % 8 == 0 ? 8 : bomb % 8);
}

void main(void)
{
	int k;
	scanf("%d", &k);
	int n;
	scanf(" %d", &n);

	func(k, n);
}

File: python 풀이/H4.py
start = int(input())
n = int(input())
time = 0
bomb = start

for i in range(n) :
  t, tpf = input().split()
  t = int(t)
  time += t
  if time >= 210 : 
    break
  else :
    if tpf == 'T' : bomb += 1
  
if bomb%8==0 : 
  print(8)
else :
  print(bomb%8)